src/main/kotlin/io/meshcloud/dockerosb/service/GenericCatalogService.kt

```diff
@@ -36,12 +32,35 @@ class CatalogConfiguration(
         return Catalog.builder().build()
       }
 
-
       val catalog = yamlHandler.readObject(statusYml, YamlCatalog::class.java)
-
       return Catalog.builder()
           .serviceDefinitions(catalog.services)
           .build()
     }
   }
+
+  fun getLatestCatalog() {
+    gitHandler.pull()
+    this.catalog = parseCatalog(gitHandler, yamlHandler)
+  }
+
+  /**
+   * When ever the endpoint /v2/catalog is accessed  it pulls the catalog from the git repo
+   */
+  override fun getCatalog(): Mono<Catalog> {
+    getLatestCatalog()
+    return Mono.just(this.catalog)
+  }
+
+  /**
+   * used to provide Catalog object to be used by other services internally
+   */
+  fun getCatalogInternal(): Catalog {
+    getLatestCatalog()
+    return this.catalog
+  }
+
+  override fun getServiceDefinition(serviceId: String?): Mono<ServiceDefinition> {
+    TODO("Not yet implemented")
```

Comment:
Is this method not called in any situation?

Reply:
As of now its not called in any situation.